inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java

```diff
@@ -298,6 +333,37 @@ private String getBackend() throws IOException {
         }
     }
 
+    @Override
+    public void snapshotState(FunctionSnapshotContext functionSnapshotContext) throws Exception {
+        Gson gson = new Gson();
+        LOG.info("snapshotState begin, context is:{}", gson.toJson(functionSnapshotContext));
+        while (numPendingRequests.get() != 0) {
+            flush();
+        }
+        if (metricData != null && metricStateListState != null) {
+            MetricStateUtils.snapshotMetricStateForSinkMetricData(metricStateListState, metricData,
+                    getRuntimeContext().getIndexOfThisSubtask());
+        }
+        LOG.info("snapshotState end, context is:{}", gson.toJson(functionSnapshotContext));
+    }
+
+    @Override
+    public void initializeState(FunctionInitializationContext context) throws Exception {
+        Gson gson = new Gson();
+        LOG.info("initializeState begin, context is:{}", gson.toJson(context));
```

Comment:
It is recommend use the json handle mode that has exists not the 'Gson' and also it is not necessary for log here?